Shared/Samples/Generate offline map with custom parameters/GenerateOfflineMapWithCustomParametersView.CustomParameters.swift

```diff
@@ -0,0 +1,189 @@
+                                Text("Min Hydrant Flow Rate")
+                                Spacer()
+                                Text(minHydrantFlowRate, format: .number.precision(.fractionLength(0)))
+                            }
+                            Slider(value: $minHydrantFlowRate, in: hydrantFlowRateRange)
+                        }
+                        
+                        HStack {
+                            Text("Crop Layer to Extent")
+                            Spacer()
+                        }
+                        GroupBox {
+                            Toggle("Water Pipes", isOn: $shouldCropWaterPipesToExtent)
+                        }
+                    }
+                }
+                .padding()
+                .toolbar {
+                    ToolbarItem(placement: .confirmationAction) {
+                        Button("Start") {
+                            dismiss()
+                            setParameterOverridesFromUI()
+                            isGeneratingOfflineMap = true
+                        }
+                    }
+                    ToolbarItem(placement: .cancellationAction) {
+                        Button("Cancel") {
+                            dismiss()
+                        }
+                    }
+                }
+                .errorAlert(presentingError: $error)
+                .task {
+                    do {
+                        // Gets the GenerateOfflineMapParameters and the
+                        // GenerateOfflineMapParameterOverrides.
+                        try await model.setUpParametersAndOverrides(extent: extent)
+                    } catch {
+                        self.error = error
+                    }
+                }
+            }
+        }
+        
+        /// Updates the `GenerateOfflineMapParameterOverrides` object with the user-set values.
+        private func setParameterOverridesFromUI() {
+            model.restrictBasemapScaleLevelRangeTo(minScaleLevel: minScaleLevel, maxScaleLevel: maxScaleLevel)
+            model.bufferBasemapAreaOfInterest(by: basemapExtentBufferDistance)
+            if !includeSystemValves {
+                model.excludeLayer(named: "System Valve")
+            }
+            if !includeServiceConnections {
+                model.excludeLayer(named: "Service Connection")
+            }
+            model.filterHydrantFlowRate(to: minHydrantFlowRate)
+            model.evaluatePipeLayersExtentCropping(for: shouldCropWaterPipesToExtent)
+        }
+    }
+}
+
+#Preview {
+    let extent = Envelope(
+        center: Point(latitude: 0, longitude: 0),
+        width: 1,
+        height: 1
+    )
+    @State var isGeneratingOfflineMap = false
+    return GenerateOfflineMapWithCustomParametersView.CustomParameters(
+        model: GenerateOfflineMapWithCustomParametersView.Model(),
+        extent: extent,
+        isGeneratingOfflineMap: $isGeneratingOfflineMap
+    )
+}
```

Comment:
Also, we generally only add previews for the top level sample views, so it might be good to remove this one. Nice attention to detail, though!